music_assistant/controllers/player_queues.py

```diff
@@ -1126,7 +1126,11 @@ async def _load_item(
                 )
             ):
                 queue_item.media_item.album = library_album
-            else:
+            elif not album and queue_item.media_item.album:
+                # Keep the good album we got from get_item_by_uri
+                pass  # Do nothing - keep existing album
+            elif album:
+                # Only restore original if we have no better alternative
```

Comment:
this entire check can be replaced with a single check to always prefer the original album
```suggestion
            elif album:
                # Restore original album if we have no better alternative from the library
```

Reply:
Thanks. Simple is best!